### DISCUSSION TITLE
Add Provider dependencies with generator and Notifier

### DISCUSSION THREAD

**Discussion by dario-digregorio** [rrousselGit/riverpod#2065] [Q&A] [answered]
Hi, I have a Notifier which I need to add dependencies. I am using riverpod generator and I could not find a way to add them there. Is there a way to add them with generator?

I get following error:
Tried to read exampleProvider:AutoDisposeNotifierProviderImpl<ExampleProvider, bool>#1927e from a place where one of its dependencies were overridden but the provider is not.

To fix this error, you can add "dependencies" to exampleProvider:AutoDisposeNotifierProviderImpl<ExampleProvider, bool>#1927e such that we have:

```
final a = Provider(...);
final b = Provider((ref) => ref.watch(a), dependencies: [a]);

**Comment by dario-digregorio** [comment]
I could add them to the generated file `example.g.dart` but I hope there is a better way. I am not very used to the meta syntax.

**Comment by rrousselGit** [comment, marked as answer]
It's work in progress. I'll be possible soon enough

**Reply by dario-digregorio** [reply]
Thank you very much!
